File: src/jdbc/MyConnection.java
package jdbc;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author jacobmenke
 */
import java.sql.*;

public class MyConnection {

    static Connection myConnection = null;
    static String url = null;

    public static Connection getConnection(String... args) {

        try {
            Class.forName("com.mysql.jdbc.Driver");
            url = "jdbc:mysql://" + new IP().getIP() +":" + new IP().getPort() + "/" + args[0] + "?user=" + args[1] + "&password=" + args[2];
            
            System.out.println(url);
            myConnection = DriverManager.getConnection(url);

        } catch (Exception e) {
            System.err.println(e);
        }

        return myConnection;

    }

}

File: src/main/kotlin/jdbc/MyConnection.kt
package jdbc

import java.sql.Connection
import java.sql.DriverManager

object MyConnection {
    @JvmStatic
    fun getConnection(vararg args: String): Connection? {
        return try {
            Class.forName("com.mysql.cj.jdbc.Driver")
            val url = "jdbc:mysql://${IP.ipAddr}:${IP.port}/${args[0]}?user=${args[1]}&password=${args[2]}"
            println(url)
            DriverManager.getConnection(url)
        } catch (e: Exception) {
            System.err.println(e)
            null
        }
    }
}
